codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSHTTPBindingProtocolGenerator.kt

```diff
@@ -81,6 +83,15 @@ abstract class AWSHTTPBindingProtocolGenerator(
         )
     }
 
+    override fun addCustomizationMiddleware(
+        ctx: ProtocolGenerator.GenerationContext,
+        operation: OperationShape,
+    ) {
+        if (ctx.service.getTrait<SigV4Trait>()?.name == "bedrock") {
+            operationMiddleware.appendMiddleware(operation, BedrockAPIKeyMiddleware())
+        }
+    }
```

Comment:
This customization point was added in https://github.com/smithy-lang/smithy-swift/pull/951.

Interceptor is installed in the stack when the signing name is `bedrock`, this is per the spec